### PR TITLE
[12.0][FIX] product_supplierinfo_for_customer: Crash when the price is computed for a template instead of variant

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT27887
ping @chienandalu @CarlosRoca13

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]
LGTM :+1:

**Comment by pedrobaeza** [comment]
Can you try to avoid code duplication making synergies between existing variant code and current one?

**Comment by sergio-teruel** [comment]
Ok.. Ready!!

**Comment by pedrobaeza** [comment]
Good job extracting the common parts. I think that you don't need a mixin as `product.product` already inherits from `product.template`. You can:

- Put all the mixin logic in `product.template`.
- Add a method there for obtaining the domain to use.
- On `product.product`, override such method for returning the changed domain.
- And that's all!

File: product_supplierinfo_for_customer/models/product_customerinfo_mixin.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        else:
+            domain = [
+                ("name", "=", partner_id),
+                "|",
+                ("product_id", "=", self.id),
+                "&",
+                ("product_tmpl_id", "=", self.product_tmpl_id.id),
+                ("product_id", "=", False),
+            ]
+        customerinfo = self.env["product.customerinfo"].search(
```
Comment (chienandalu) [inline product_supplierinfo_for_customer/models/product_customerinfo_mixin.py]:
```suggestion
        customerinfo = self.env["product.customerinfo"].sudo().search(
```

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Is this needed for v13?

/ocabot merge patch

**Pull request merged**